### ISSUE TITLE
Expandable MaterialSideNavMini does not work with MaterialCollapsible

### ISSUE THREAD

**Issue by chamgda** [GwtMaterialDesign/gwt-material#902]
The demo page for SideNavs makes it seem like you can use MaterialCollapsible in any type of SideNav to create submenus.

I tried to do that with an expandable MaterialSideNavMini and everything is added to the DOM as expected. But when I click the CollapsibleItemHeader (MaterialLink inside), nothing happens. I couldn't figure out what causes this but it works if I manually add a ClickHandler to the CollapsibleItemHeader (the MaterialCollapsible itself is part of a ui.xml):

```
    MaterialCollapsibleItem item = new MaterialCollapsibleItem();
    MaterialIcon icon = iconType != null ? new MaterialIcon(iconType) : null;
    MaterialCollapsibleHeader header = new MaterialCollapsibleHeader(new MaterialLink(label, icon));
    item.add(header);
    header.addClickHandler(event -> {
        if (item.isActive()) {
            item.collapse();
        } else {
            item.expand();
        }
    });
    UnorderedList ul = new UnorderedList();
    MaterialCollapsibleBody body = new MaterialCollapsibleBody(ul);
    item.add(body);
```

I guess you could make the argument that expandable mini SideNavs are not supposed to contain collapsibles, it's possible though and in my oppinion putting them into a SideNavMini shouldn't change how collapsibles behave.

Default behaviour when collapsing the SideNav could be to simply collapse all CollapsibleItems as well (at least that's what I did).

**Comment by kevzlou7979** [comment]
We could make this as feature request to support MaterialSideNavMini with MaterialCollapsible this could also be glued to https://github.com/GwtMaterialDesign/gwt-material/issues/913

**Comment by chamgda** [comment]
Well if I understand the feature request in #913 correctly, he is talking about using MaterialCollapsibles as standard MaterialCollapsibles in a SideNav. My use case is using MaterialCollapsibles as expandable menu items (the way it is implemented right now), which doesn't work correctly in expandable MaterialSideNavMinis. Those two are pretty conflicting... Are you planning on adding a flag or something similar to mark if a collapsible should be displayed in the standard way or as a menu item in a SideNav? Depending on how this is handled, this could be a major breaking change...

**Comment by kevzlou7979** [comment]
Yeah youre right, we could add a flag to collapsible module, maybe a displayType or something that could force it to be  a standard one or a sidenav nested collapsible,

**Comment by kevzlou7979** [comment]
We could potentially have:

collapsible#setDisplayType(DEFAULT, SIDENAV)

**Comment by chamgda** [comment]
Both options seem fine to me.

I like the first option because it offers more flexibility.

I like the second option because it conforms pretty much exactly to the contract of MaterialSideNavContent as described in its JavaDocs.

**Comment by kevzlou7979** [comment]
Ok will be adding this on our backlog for 2.3
